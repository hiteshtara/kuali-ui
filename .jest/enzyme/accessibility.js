/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
