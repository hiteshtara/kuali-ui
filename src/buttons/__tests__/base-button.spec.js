/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { mount, shallow } from 'enzyme'

import { Icon } from '../..'
import BaseButton from '..'

describe('BaseButton Component', () => {
  it('should render', () => {
    const wrapper = shallow(<BaseButton />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with an Icon', () => {
    const wrapper = shallow(
      <BaseButton>
        <Icon name='done' />
      </BaseButton>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with a string icon name', () => {
    const wrapper = shallow(<BaseButton>done</BaseButton>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should preserve the `className` prop on an Icon child', () => {
    const customClassName = 'the-custom-class'

    const wrapper = mount(
      <BaseButton>
        <Icon name='add' className={customClassName} />
      </BaseButton>
    )

    expect(wrapper.find(Icon).hasClass(customClassName)).toBeTruthy()
  })

  it('should apply a ref callback to button node via innerRef prop', () => {
    const spy = jest.fn()
    const wrapper = mount(<BaseButton innerRef={spy} />)
    expect(spy).toHaveBeenCalled()
    const button = wrapper.find('button').instance()
    expect(spy).toHaveBeenLastCalledWith(button)
  })
})
