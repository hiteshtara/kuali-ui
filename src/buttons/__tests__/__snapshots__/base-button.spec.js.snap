// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton Component should render 1`] = `
<withTooltip(Button)
  block={false}
  component="button"
  icon={false}
  iconBefore={true}
  ink={
    <InkContainer
      className={undefined}
      disabledInteractions={undefined}
      inkClassName={undefined}
      inkStyle={undefined}
      pulse={undefined}
      style={undefined}
      transitionEnterTimeout={450}
      transitionLeaveTimeout={300}
      transitionOverlap={150}
      waitForInkTransition={undefined}
    />
  }
  size="default"
  type="submit"
  variant="default"
/>
`;

exports[`BaseButton Component should render with a string icon name 1`] = `
<withTooltip(Button)
  block={false}
  component="button"
  icon={false}
  iconBefore={true}
  ink={
    <InkContainer
      className={undefined}
      disabledInteractions={undefined}
      inkClassName={undefined}
      inkStyle={undefined}
      pulse={undefined}
      style={undefined}
      transitionEnterTimeout={450}
      transitionLeaveTimeout={300}
      transitionOverlap={150}
      waitForInkTransition={undefined}
    />
  }
  size="default"
  type="submit"
  variant="default"
>
  done
</withTooltip(Button)>
`;

exports[`BaseButton Component should render with an Icon 1`] = `
<withTooltip(Button)
  block={false}
  component="button"
  icon={false}
  iconBefore={true}
  ink={
    <InkContainer
      className={undefined}
      disabledInteractions={undefined}
      inkClassName={undefined}
      inkStyle={undefined}
      pulse={undefined}
      style={undefined}
      transitionEnterTimeout={450}
      transitionLeaveTimeout={300}
      transitionOverlap={150}
      waitForInkTransition={undefined}
    />
  }
  size="default"
  type="submit"
  variant="default"
>
  <Icon
    aria-hidden={true}
    iconClassName="material-icons"
    name="done"
    variant="default"
  />
</withTooltip(Button)>
`;
